address_book.py

```diff
@@ -10,57 +10,60 @@ def __init__(self, name, email_address, phone_number):
 
     def show(self):
         """Shows details."""
-        print(f'Name = {self.name}, Email Address = {self.email_address}, Phone number = {self.phone_number}'
+        print(
+            f'Name = {self.name}, Email Address = {self.email_address}, Phone number = {self.phone_number}')
 
 
 class AddressBook:
-    def __init__(self):
-        self.contacts = {}
+    """Write a name for your file as the argument for the AddressBook you are instantiating. 
+       This will be the file where the contacts are saved. .json will be added to the filename automatically."""
 
-    def add_contact(self, contact, object):
+    def __init__(self, contacts_file):
+
+        # This is where the data will be saved
+        self.contacts_file = contacts_file
+        self._contacts = {}
+
+        # if the file already exists, load the data in into the _contacts dict by iterating through the file
+        if Path(self.contacts_file).exists == True:
+            with open(self.contacts_file) as f:
+                data = json.load(f)
+
+            for name, details in data.items():
+                self._contacts[name] = Contact(
+                    name, details['email_address'], details['phone_number'])
+
+    def add_contact(self, contact):
         """add contact to address book, contact is the key you will use to find the contact and
          the value is the contact object"""
-        self.contacts[contact] = object
-
-        # Name of the file that will store the address book object
-        # Write to the file
-        f = open(contactsfile, 'wb')
-        # Dump to the file
-        pickle.dump(self.contacts, f)
-        f.close()
-        print("Added Contact", contact)
+        self._contacts[contact.name] = vars(contact)
+        print(f"Added Contact {contact.name}")
 
     def remove_contact(self, contact):
         """Removes contact"""
-        if contact in self.contacts:
-            del self.contacts[contact]
-
-            # Name of the file that will store the address book object
-            # Write to the file
-            f = open(contactsfile, 'wb')
-            # Dump to the file
-            pickle.dump(self.contacts, f)
-            f.close()
+        if contact in self._contacts:
+            del self._contacts[contact]
             print("Removed Contact", contact)
 
         else:
             print("Could not delete contact, contact does not exist.")
 
     def search_contact(self, contact):
-        """searches the dictionary for the key passed in as an arg and returns the Contact object's detail's"""
-        if contact in self.contacts:
-            print(self.contacts.get(contact).show())
+        """ Searches the dictionary for the key passed in as an arg and returns the Contact object's detail's. """
+        if contact in self._contacts:
+            print(self._contacts.get(contact))
 
         else:
             print("Contact not found.")
```

Comment:
Again - get rid of these print statements. Let the user interface handle how and what to show the user. This class is a collection of data with methods for handling that, not really user interaction. Return nothing is fine here.